assert/assertion_format.go

```diff
@@ -618,10 +618,11 @@ func NotSamef(t TestingT, expected interface{}, actual interface{}, msg string,
 	return NotSame(t, expected, actual, append([]interface{}{msg}, args...)...)
 }
 
-// NotSubsetf asserts that the specified list(array, slice...) contains not all
-// elements given in the specified subset(array, slice...).
+// NotSubsetf asserts that the specified list(array, slice...) or map contains not all
+// elements given in the specified subset list(array, slice...) or map.
 //
 //    assert.NotSubsetf(t, [1, 3, 4], [1, 2], "But [1, 3, 4] does not contain [1, 2]", "error message %s", "formatted")
+//    assert.NotSubsetf(t, {"x": 1, "y": 2}, {"z": 3}, "But {"x": 1, "y": 2} does not contain {"z": 3}", "error message %s", "formatted")
```

Comment:
The problem being that using static values for the example means that you cannot provide an example that would pass the assertion and include the values used in the example as part of the failure string - because that string would never be output since the example would always pass.  However, the string is still incorrectly inverted because the string would only be output if arg1 DOES contain arg2 rather than "does not"

But then, it doesn't ever make sense to include the actual args statically in the failure string, anyway, since the behaviour of the package is already to include the parameters as part of the output.  So it might actually be better for the example to literally include the string `But arg1 does contain arg2` in order to eliminate the confusion

Reply:
Removed the message string here - no other examples in the library do this, so to be consistent I've removed it from both the slice and the map examples.